source/renderer/app/components/wallet/summary/WalletSummaryHeaderRewards.js

```diff
@@ -0,0 +1,92 @@
+// @flow
+import BigNumber from 'bignumber.js';
+import React from 'react';
+import { observer } from 'mobx-react';
+import {
+  defineMessages,
+  FormattedHTMLMessage,
+  FormattedMessage,
+} from 'react-intl';
+import { PopOver } from 'react-polymorph/lib/components/PopOver';
+import SVGInline from 'react-svg-inline';
+import questionMarkIcon from '../../../assets/images/question-mark.inline.svg';
+import { useDiscreetModeFeature } from '../../../features';
+import type { ReplacerFn } from '../../../features/discreet-mode/types';
+import { formattedWalletAmount } from '../../../utils/formatters';
+import styles from './WalletSummaryHeaderRewards.scss';
+
+const messages = defineMessages({
+  rewards: {
+    id: 'wallet.summary.header.rewardsSummary',
+    defaultMessage: '!!!{total} rewards earned, {unspent} unspent rewards',
+    description: 'Headline for the Decentralisation notification.',
+  },
+  unspendableTooltip: {
+    id: 'wallet.summary.header.unspendableTooltip',
+    defaultMessage:
+      '!!!info message about unspendable rewards due to missing utxos',
+    description:
+      'Tooltip describing that rewards are unspendable on the Wallet summary header',
+  },
+});
+
+export function getFormattedRewardAmount(amount: BigNumber): string {
+  return amount.isGreaterThan(0) && amount.isLessThan(0.1)
+    ? '< 0.1 ADA'
+    : formattedWalletAmount(amount, true, false, 'ADA', 1);
+}
+
+export function discreetRewardsAmount(
```

Comment:
No, i think it would make the implementation just worse because there are some special cases for rewards and tokens that are irrelevant to the other … the replacer API was designed to be used for various use cases so i would keep it like that